tests/need_credentials/test_azure_guidance.py

```diff
@@ -15,6 +18,93 @@ def azure_guidance_model(selected_model, selected_model_name):
         pytest.skip("Requires Azure Guidance model")
 
 
+def test_azure_guidance_fill_in_json(azure_guidance_model: guidance.models.Model):
+    lm = azure_guidance_model
+    @guidance(stateless=True, dedent=False)
+    def character_maker(lm, id, description, valid_weapons):
+        lm += f"""\
+        The following is a short character profile for an RPG game in JSON format.
+        ```json
+        {{
+            "id": "{id}",
+            "description": "{description}",
+            "name": "{gen('name', stop='"')}",
+            "age": {gen('age', regex='[0-9]+', stop=',')},
+            "armor": "{select(options=['leather', 'chainmail', 'plate'], name='armor')}",
+            "weapon": "{select(options=valid_weapons, name='weapon')}",
+            "class": "{gen('class', stop='"')}",
+            "mantra": "{gen('mantra', stop='"')}",
+            "strength": {gen('strength', regex='[0-9]+', stop=',')},
+            "items": ["{gen('item', list_append=True, stop='"')}", "{gen('item', list_append=True, stop='"')}", "{gen('item', list_append=True, stop='"')}"]
+        }}```"""
+        return lm
+    lm += character_maker(1, 'A nimble fighter', ['axe', 'sword', 'bow'])
+    result = str(lm)
+    json_text = result[result.find("```json") + 8:-3]
+    json.loads(json_text)  # check for valid JSON
+
+
+def test_azure_guidance_basic(azure_guidance_model: guidance.models.Model):
+    lm = azure_guidance_model
+    lm += "Write a number: " + gen("text", max_tokens=3)
+    assert len(lm["text"]) >= 3
+
+def test_azure_guidance_56(azure_guidance_model: guidance.models.Model):
+    lm = azure_guidance_model
+    # make sure we recognize EOS token correctly
+    lm += "Q: 7 * 8\nA: " + gen("text", regex="[0-9]+", max_tokens=20)
+    assert lm["text"] == "56"
+
+def test_azure_guidance_56_newline(azure_guidance_model: guidance.models.Model):
+    lm = azure_guidance_model
+    # make sure we recognize EOS token correctly
+    lm += "Q: 7 * 8\nA: " + gen("text", regex="[0-9]+", max_tokens=20) + "\n"
+    assert lm["text"] == "56"
+
+def test_azure_guidance_1003(azure_guidance_model: guidance.models.Model):
+    lm = azure_guidance_model
+    lm += "Q: 1000 + 3\nA: " + gen("text", regex="[0-9]+", max_tokens=20)
+    assert lm["text"] == "1003"
+
+def test_azure_guidance_dolphins(azure_guidance_model: guidance.models.Model):
+    lm = azure_guidance_model
+    # Yes|No has an implicit forced EoS at the end, which should not be actually generated
+    lm += "Q: Are dolphins fish?\nA: " + gen("dolphins", regex="Yes|No", max_tokens=10) + \
+        "\nQ: Are salmons fish?\nA: " + gen("sharks", regex="Yes|No", max_tokens=10)
```

Comment:
you probably meant `gen("salmon"` instead of `gen("sharks"`

Reply:
😆 @mmoskal

Reply:
it used to be sharks, but somehow some Phi versions didn't think they were fish...